src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -1225,12 +1225,15 @@ private Binder GetParameterNameAttributeValueBinder(MemberDeclarationSyntax memb
                     _ = typeDeclaration.ParameterList;
                     Binder outerBinder = VisitCore(memberSyntax);
                     SourceNamedTypeSymbol type = ((NamespaceOrTypeSymbol)outerBinder.ContainingMemberOrLambda).GetSourceTypeMember((TypeDeclarationSyntax)memberSyntax);
-                    var primaryConstructor = type.PrimaryConstructor;
-
-                    if (primaryConstructor.SyntaxRef.SyntaxTree == memberSyntax.SyntaxTree &&
-                        primaryConstructor.GetSyntax() == memberSyntax)
+                    if (type is not null)
```

Comment:
> if (type is not null) [](http://example.com/codeflow?start=19&length=22)

While this change, perhaps, eliminates the crash for the specific scenario, it looks to me that ```GetAssociatedMemberForXmlSyntax``` shouldn't be returning type syntax in this case. This is not a doc comment on the type declaration. The binder thinks we are inside the type `C` and we trying to locate declaration of the same `C` within it. The search, of course, fails. It is quite possible that other code paths (other XML tags) in ```VisitXmlNameAttribute``` are negatively affected by the ```GetAssociatedMemberForXmlSyntax``` behavior as well.
To summarize, I think the change is fixing a symptom rather thana root cause of the problem.

Reply:
It looks like some other scenarios rely on this behavior. For example:

[https://github.com/dotnet/roslyn/blob/40d3f69e6d2e2d021c2ad546734ec6fb8d7c82d8/src/Compilers/CSharp/Test/Symbol/DocumentationComments/ParameterTests.cs#L760-L833](https://github.com/dotnet/roslyn/blob/40d3f69e6d2e2d021c2ad546734ec6fb8d7c82d8/src/Compilers/CSharp/Test/Symbol/DocumentationComments/ParameterTests.cs#L760-L833)

Reply:
> It looks like some other scenarios rely on this behavior. 

Could you please elaborate? It doesn't look like the test is using primary constructors in any form and this code path is specific to primary constructors.

Reply:
I was referring to the behavior I tried to implement which you described in the comment above:

> it looks to me that `GetAssociatedMemberForXmlSyntax` shouldn't be returning type syntax in this case. This is not a doc comment on the type declaration. The binder thinks we are inside the type `C` and we trying to locate declaration of the same `C` within it

That's not specific to primary constructors, right?